src/plugins/preauth/pkinit/pkinit_crypto_openssl.c

```diff
@@ -5932,7 +5932,7 @@ pkcs7_dataDecode(krb5_context context,
     }
     if (ek != NULL) {
         OPENSSL_cleanse(ek, eklen);
-        OPENSSL_free(ek);
+        free(ek);
```

Comment:
There is a (minor) problem here, but the fix isn't so simple, because ek could be replaced with tkey, which is allocated with OPENSSL_malloc().  I think there's also a memory leak at line 5903.

We can change tkey to be allocated normally, and use zapfree() instead of OPENSSL_cleanse() and OPENSSL_free() for both values.

Out of curiosity, what tool flagged this?

Reply:
I believe it is Coverity - from the ALLOC_FREE_MISMATCH checker.